torchtune/modules/transformer.py

```diff
@@ -36,6 +36,23 @@ def __init__(
         self.mlp_norm = mlp_norm
         self.mlp = mlp
 
+    def setup_cache(self, batch_size: int, dtype: torch.dtype) -> None:
```

Comment:
Maybe a controversial q but if we are gonna have multiple different types of layers that can be passed to `TransformerDecoder` should we define some kind of Protocol to ensure any individual layer has this method defined? (I don't have a strong opinion on this tbh, just putting it out there)

Reply:
In a early version of the TransformerDecoderLayer RFC I had a Protocol and generally got poor reception. But I'm open to adding it back in.

Reply:
Lol oh yeah I remember that now. Let's leave it like this